drivers/spi/spi_andes_atcspi200.c

```diff
@@ -0,0 +1,629 @@
+	uint32_t tx_fifo_size;
+	uint32_t rx_fifo_size;
+	int tx_cnt;
+	uint32_t is_cmdaddr_mode;
+	size_t chunk_len;
+	bool busy;
+};
+
+struct spi_atcspi200_cfg {
+	atcspi200_cfg_func_t cfg_func;
+	uint32_t base;
+	uint32_t irq_num;
+	uint32_t f_sys;
+	bool xip;
+};
+
+/* API Functions */
+static int spi_config(const struct device *dev,
+		      const struct spi_config *config)
+{
+	const struct spi_atcspi200_cfg * const cfg = DEV_CFG(dev);
+	uint32_t sclk_div, data_len;
+
+	/* Set the divisor for SPI interface sclk */
+	sclk_div = (cfg->f_sys / (config->frequency << 1)) - 1;
+	CLR_MASK(SPI_TIMIN(dev), REG_TIMIN_SCLK_DIV_MSK);
+	SET_MASK(SPI_TIMIN(dev), sclk_div);
+
+	/* Set Master mode */
+	CLR_MASK(SPI_TFMAT(dev), REG_TFMAT_SLVMODE_MSK);
+
+	/* Disable data merge mode */
+	CLR_MASK(SPI_TFMAT(dev), REG_TFMAT_DATA_MERGE_MSK);
+
+	/* Set data length */
+	data_len = SPI_WORD_SIZE_GET(config->operation) - 1;
+	CLR_MASK(SPI_TFMAT(dev), REG_TFMAT_DATA_LEN_MSK);
+	SET_MASK(SPI_TFMAT(dev), (data_len << REG_TFMAT_DATA_LEN_OFFSET));
+
+	/* Set SPI frame format */
+	if (config->operation & SPI_MODE_CPHA)
+		SET_MASK(SPI_TFMAT(dev), REG_TFMAT_CPHA_MSK);
+	else
+		CLR_MASK(SPI_TFMAT(dev), REG_TFMAT_CPHA_MSK);
+
+	if (config->operation & SPI_MODE_CPOL)
+		SET_MASK(SPI_TFMAT(dev), REG_TFMAT_CPOL_MSK);
+	else
+		CLR_MASK(SPI_TFMAT(dev), REG_TFMAT_CPOL_MSK);
+
+	/* Set SPI bit order */
+	if (config->operation & SPI_TRANSFER_LSB)
+		SET_MASK(SPI_TFMAT(dev), REG_TFMAT_LSB_MSK);
+	else
+		CLR_MASK(SPI_TFMAT(dev), REG_TFMAT_LSB_MSK);
+
+	/* Set TX/RX FIFO threshold */
+	CLR_MASK(SPI_CTRL(dev), REG_CTRL_TX_THRES_MSK);
+	CLR_MASK(SPI_CTRL(dev), REG_CTRL_RX_THRES_MSK);
+
+	SET_MASK(SPI_CTRL(dev), TX_FIFO_THRESHOLD << REG_CTRL_TX_THRES_OFFSET);
+	SET_MASK(SPI_CTRL(dev), RX_FIFO_THRESHOLD << REG_CTRL_RX_THRES_OFFSET);
+
+	return 0;
+}
+
+static int spi_transfer(const struct device *dev, uint32_t len)
+{
+
+	struct spi_atcspi200_data * const data = DEV_DATA(dev);
+	struct spi_context *ctx = &data->ctx;
+	uint32_t data_len, tctrl, int_msk;
+
+	if (len > MAX_TRANSFER_CNT) {
+		return -EINVAL;
+	}
+
+	data_len = len - 1;
+	data->tx_cnt = 0;
+
+	if (!spi_context_rx_on(ctx)) {
+		tctrl = (TRNS_MODE_WRITE_ONLY << REG_TCTRL_TRNS_MODE_OFFSET) |
+			(data_len << REG_TCTRL_WR_TCNT_OFFSET);
+		int_msk = REG_IEN_TX_FIFO_MSK | REG_IEN_END_MSK;
+	} else if (!spi_context_tx_on(ctx)) {
+		tctrl = (TRNS_MODE_READ_ONLY << REG_TCTRL_TRNS_MODE_OFFSET) |
+			(data_len << REG_TCTRL_RD_TCNT_OFFSET);
+		int_msk = REG_IEN_RX_FIFO_MSK | REG_IEN_END_MSK;
+	} else {
+		tctrl = (TRNS_MODE_WRITE_READ << REG_TCTRL_TRNS_MODE_OFFSET) |
+			(data_len << REG_TCTRL_WR_TCNT_OFFSET) |
+			(data_len << REG_TCTRL_RD_TCNT_OFFSET);
+		int_msk = REG_IEN_TX_FIFO_MSK |
+			  REG_IEN_RX_FIFO_MSK |
+			  REG_IEN_END_MSK;
+	}
+
+	sys_write32(tctrl, SPI_TCTRL(dev));
+
+	/* Enable TX/RX FIFO interrupts */
+	sys_write32(int_msk, SPI_INTEN(dev));
+
+	if (!data->is_cmdaddr_mode) {
+		/* Start transfering */
```

Comment:
`transfering` -> `transferring` (checkpatch complains about this typo)

Reply:
Thanks for reviewing!!
I have fixed it.